src/docdb/tree/DocDBStoredProcedureTreeItem.ts

```diff
@@ -53,17 +62,8 @@ export class DocDBStoredProcedureTreeItem extends AzureTreeItem<IDocDBTreeRoot>
 
     public async writeFileContent(_context: IActionContext, content: string): Promise<void> {
         const client = this.root.getDocumentClient();
-        this.procedure = await new Promise<ProcedureMeta>((resolve, reject) => client.replaceStoredProcedure(
-            this.link,
-            { body: content, id: this.procedure.id },
-            (err, updated: ProcedureMeta) => {
-                if (err) {
-                    reject(err);
-                } else {
-                    resolve(updated);
-                }
-            })
-        );
+        const replace = await client.database(this.parent.parent.parent.id).container(this.parent.parent.id).scripts.storedProcedure(this.id).replace({ id: this.id, body: content });
```

Comment:
I think it's too easy to accidentally use the wrong parent id when you're doing this kind of stuff. How about at each tree-level you make a method like `getDatabaseClient` and `getContainerClient`. I imagine `getContainerClient` would be something like this:
```
return (await this.parent.getDatabaseClient()).container(this.id);
```

The benefit is that you're only specifying 'id' within that method, so if you get it wrong it would affect all usages and we would find out faster.